python-package/lightgbm/sklearn.py

```diff
@@ -586,13 +589,16 @@ def _get_meta_data(collection, name, i):
                                                    valid_weight, valid_init_score, valid_group, params)
                 valid_sets.append(valid_set)
 
+        if init_model is not None and isinstance(init_model, LGBMModel):
```

Comment:
Seems that the `init_model is not None` check is redundant. `isinstance(init_model, LGBMModel)` is enough.